### PR TITLE
Fixes #27996 - Add template for entitlements report

### PR DESCRIPTION
**Pull request by xprazak2**
Needs:
https://github.com/theforeman/foreman/pull/7081
https://github.com/Katello/katello/pull/8369

### CODE REVIEW

File: report_templates/entitlements.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+name: Entitlements
+snippet: false
+model: ReportTemplate
+require:
+- plugin: katello
+  version: 3.14.0
+%>
+
+<%- load_hosts(includes: [:lifecycle_environment, :operatingsystem, :architecture, :content_view, :organization, :reported_data, :subscription_facet, :pools => [:subscription]]).each_record do |host| -%>
+  <%- host.pools.each do |pool| -%>
```
Comment (ares) [inline report_templates/entitlements.erb]: usually we do the indentation inside of erb tag, see https://github.com/theforeman/community-templates/blob/develop/report_templates/applicable_errata.erb#L24-L26 for example

File: report_templates/entitlements.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          'Lifecycle Environment': host.lifecycle_environment,
+          'Content View': host.content_view,
+          'Host Collections': host.host_collections,
+          'Virtual': host.virtual,
+          'Guest of Host': host.hypervisor_host,
+          'OS': host.operatingsystem,
+          'Arch': host.architecture,
+          'Sockets': host.sockets,
+          'RAM': host.ram,
+          'Cores':  host.cores,
```
Comment (ares) [inline report_templates/entitlements.erb]: two spaces after colon

File: report_templates/entitlements.erb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          'Subscription Contract': pool.contract_number,
+          'Subscription Account': pool.account_number,
+          'Subscription Start': pool.start_date,
+          'Subscription End': pool.end_date,
+          'Subscription Guest': registered_through(host)
+          )
+      -%>
+    <%- end -%>
+<%- end -%>
+
```
Comment (ares) [inline report_templates/entitlements.erb]: I think this would cause extra empty line e.g. in csv output

File: report_templates/entitlements.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<%#
+name: Entitlements
+snippet: false
+model: ReportTemplate
+require:
+- plugin: katello
+  version: 3.14.0
+%>
+
```
Comment (ares) [inline report_templates/entitlements.erb]: I think this would cause extra empty line, ideally also add - to the comment closing erb tag

**Review by ares — CHANGES_REQUESTED** [review]
there are few "rubocoppy" nits, but looks really great! of course can only be merged after dependencies are in, but looking forward to merge this one :-)

**Comment by xprazak2** [comment]
Updated with style fixes.

**Review by ares — COMMENTED** [review]
looks great, waiting on depending PRs before testing this

**Review by ares — APPROVED** [review]
Tested with Katello PR (foreman is in already) and it's working just fine, ready to be merged after Katello PR gets in.

**Comment by ares** [comment]
Thanks @xprazak2, merging!

**Pull request merged**